stdlib/public/RemoteInspection/TypeRefBuilder.cpp

```diff
@@ -348,15 +348,21 @@ RemoteRef<FieldDescriptor> TypeRefBuilder::getFieldTypeInfo(const TypeRef *TR) {
     }
   }
 
-  // On failure, fill out the cache, ReflectionInfo by ReflectionInfo,
-  // until we find the field descriptor we're looking for.
+  // If the heuristic didn't work, iterate over every reflection info
+  // that the external cache hasn't processed.
   for (size_t i = 0; i < ReflectionInfos.size(); ++i) {
     if (ExternalTypeRefCache && ExternalTypeRefCache->isReflectionInfoCached(i))
       continue;
     if (auto FD = findFieldDescriptorAtIndex(i, *MangledName))
       return *FD;
   }
 
+  // If we still haven't found the field descriptor, go over every reflection
+  // info, even the ones the external cache supposedly processed.
```

Comment:
Can you clarify whether this is intended as a workaround for a specific issue, or whether the contract for ExternalTypeRefCache is that it should only serve as a positive cache and not cache negative lookups?

Reply:
The initial motivation was as a workaround to a specific bug, but I'm thinking that this should be the behavior even we solve the other bug. What do you think?

Reply:
Either behavior is fine, I just wanted to make sure we explicitly document the contract in the code.